inst/private/doctest_collect.m

```diff
@@ -125,6 +125,19 @@
 end
 
 
+% Deal with .oct files, which may define more than one function
+if (strcmp(type, 'octfile'))
+  len = numel (what) + 4;
+  list = autoload ();
+  pmatch = @(e) (numel (e.file) > len) && strcmp (e.file(end-len+1:end), [what '.oct']);
```

Comment:
I think this only matches the name of the oct file: what if an oct file has same name in a different path?  In my patch, I used `pwd/what` but I'm not sure that's right either---should `doctest meh.oct` work if meh.oct is in the path but not in pwd?  Maybe your way is less risky...

Reply:
Yes, this only matches the basename of the .oct file.

I didn't want to use `pwd` because the user could be testing any .oct file in the load path, right?

It's true that this is a weaker check, a stronger check might be to get the full path to the .oct file with `which` or `file_in_loadpath`?